frontend/src/metabase/visualizations/shared/settings/cartesian-chart.ts

```diff
@@ -0,0 +1,74 @@
+import _ from "underscore";
+import type { ComputedVisualizationSettings } from "metabase/visualizations/types";
+import type { Card, SeriesOrderSetting } from "metabase-types/api";
+
+export const getDefaultStackingValue = (
+  settings: ComputedVisualizationSettings,
+  card: Card,
+) => {
+  // legacy setting and default for D-M-M+ charts
+  if (settings["stackable.stacked"]) {
+    return settings["stackable.stacked"];
+  }
+
+  const shouldStack =
+    card.display === "area" &&
+    ((settings["graph.metrics"] ?? []).length > 1 ||
+      (settings["graph.dimensions"] ?? []).length > 1);
+
+  return shouldStack ? "stacked" : null;
+};
+
+export const getSeriesOrderVisibilitySettings = (
+  settings: ComputedVisualizationSettings,
+  seriesKeys: string[],
+) => {
+  const seriesSettings = settings["series_settings"];
+  const seriesColors = settings["series_settings.colors"] || {};
+  const seriesOrder = settings["graph.series_order"];
+  // Because this setting is a read dependency of graph.series_order_dimension, this should
+  // Always be the stored setting, not calculated.
+  const seriesOrderDimension = settings["graph.series_order_dimension"];
+  const currentDimension = settings["graph.dimensions"]?.[1];
+
+  if (currentDimension === undefined) {
+    return [];
+  }
+
+  const generateDefault = (keys: string[]) => {
+    return keys.map(key => ({
+      key,
+      color: seriesColors[key],
+      enabled: true,
+      name: seriesSettings?.[key]?.title || key,
+    }));
+  };
+
+  const removeMissingOrder = (keys: string[], order: SeriesOrderSetting[]) =>
+    order.filter(o => keys.includes(o.key));
+  const newKeys = (keys: string[], order: SeriesOrderSetting[]) =>
```

Comment:
```suggestion
  const getNewKeys = (keys: string[], order: SeriesOrderSetting[]) =>
```

Name should start with a verb to indicate it's a function, not precomputed values.